src/jit_channel/channel_manager.rs

```diff
@@ -0,0 +1,880 @@
+	OM::Target: OnionMessageHandler,
+	CMH::Target: CustomMessageHandler,
+	NS::Target: NodeSigner,
+{
+	pub(crate) fn new(
+		entropy_source: ES, promise_secret: [u8; 32],
+		pending_messages: Arc<Mutex<Vec<(PublicKey, LSPSMessage)>>>,
+		pending_events: Arc<EventQueue>,
+		channel_manager: Arc<ChannelManager<M, T, ES, NS, SP, F, R, L>>,
+	) -> Self {
+		Self {
+			entropy_source,
+			promise_secret,
+			pending_messages,
+			pending_events,
+			per_peer_state: RwLock::new(HashMap::new()),
+			channels_by_scid: RwLock::new(HashMap::new()),
+			peer_manager: Mutex::new(None),
+			channel_manager,
+		}
+	}
+
+	pub fn set_peer_manager(
+		&self, peer_manager: Arc<PeerManager<Descriptor, CM, RM, OM, L, CMH, NS>>,
+	) {
+		*self.peer_manager.lock().unwrap() = Some(peer_manager);
+	}
+
+	pub fn create_invoice(
+		&self, counterparty_node_id: PublicKey, payment_size_msat: Option<u64>,
+		token: Option<String>, user_channel_id: u128,
+	) {
+		let channel_id = self.generate_channel_id();
+		let channel = JITChannel::new(
+			channel_id,
+			counterparty_node_id,
+			user_channel_id,
+			payment_size_msat,
+			token,
+		);
+
+		let mut per_peer_state = self.per_peer_state.write().unwrap();
+		let peer_state_mutex =
+			per_peer_state.entry(counterparty_node_id).or_insert(Mutex::new(PeerState::default()));
+		let peer_state = peer_state_mutex.get_mut().unwrap();
+		peer_state.insert_channel(channel_id, channel);
+
+		let request_id = self.generate_request_id();
+		peer_state.insert_request(request_id.clone(), channel_id);
+
+		{
+			let mut pending_messages = self.pending_messages.lock().unwrap();
+			pending_messages.push((
+				counterparty_node_id,
+				Message::Request(request_id, Request::GetVersions(GetVersionsRequest {})).into(),
+			));
+		}
+
+		if let Some(peer_manager) = self.peer_manager.lock().unwrap().as_ref() {
+			peer_manager.process_events();
+		}
+	}
+
+	pub fn opening_fee_params_generated(
+		&self, counterparty_node_id: PublicKey, request_id: RequestId,
+		opening_fee_params_menu: Vec<RawOpeningFeeParams>, min_payment_size_msat: u64,
+		max_payment_size_msat: u64,
+	) -> Result<(), APIError> {
+		let per_peer_state = self.per_peer_state.read().unwrap();
+
+		match per_peer_state.get(&counterparty_node_id) {
+			Some(peer_state_mutex) => {
+				let mut peer_state = peer_state_mutex.lock().unwrap();
+
+				match peer_state.pending_requests.remove(&request_id) {
+					Some(Request::GetInfo(_)) => {
+						let response = Response::GetInfo(GetInfoResponse {
+							opening_fee_params_menu: opening_fee_params_menu
+								.into_iter()
+								.map(|param| param.into_opening_fee_params(&self.promise_secret))
+								.collect(),
+							min_payment_size_msat,
+							max_payment_size_msat,
+						});
+						self.enqueue_response(counterparty_node_id, request_id, response);
+						Ok(())
+					}
+					_ => Err(APIError::APIMisuseError {
+						err: format!(
+							"No pending get_info request for request_id: {:?}",
```

Comment:
nit: Impl `Display` for `RequestId` and use
```suggestion
							"No pending get_info request for request_id: {}",
```